### PR TITLE
chore: Update ai-api clients

### PR DESCRIPTION
**Pull request by shibeshduw**
## Context

Generated ai-api clients with v2.34.0 of the spec as per: https://github.wdf.sap.corp/AI/apis/blob/master/aicore.yaml
Also aligns the clients with the spec that Java SDK is using.
Finally, a regeneration was required after AI/gen-ai-hub-sdk-js-backlog#73.

There are minor type updates which do not affect any of our existing test cases. 

## Definition of Done

- [x] Code is tested (Unit, E2E)
- [x] Error handling created / updated & covered by the tests above
- ~[ ] Documentation updated~
- [x] (Optional) Aligned changes with the Java SDK
- [x] (Optional) Release notes updated

### CODE REVIEW

**Review by ZhongpinWang — APPROVED** [review]
LGTM

**Comment by deekshas8** [comment]
@shibeshduw I see deletion of one api, this is a breaking change now. I discussed this with @marikaner . Please add a compatibility changelog for this. It would also make sense to mention to users that this package contains generated code and therefore updates to it can have breaking changes. We can advise users to update this specific dependency with `~` instead of `^` (in README). @jjtang1985 

There are also some new functions/types (incl. a fix in `AiExecutionModificationResponseList`). Maybe have a changelog for these as well

**Comment by jjtang1985** [comment]
> Please add a compatibility changelog for this. It would also make sense to mention to users that this package contains generated code and therefore updates to it can have breaking changes.

Breaking changes from spec is fine, if we document that.
Then we forward the issue to the spec owner.

File: .changeset/tiny-pears-walk.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-ai-sdk/ai-api': minor
+---
+
+[Fixed Issue] Adjust `AiExecutionModificationResponseList` type to correctly represent an array of responses or errors.
```
Comment (deekshas8) [inline .changeset/tiny-pears-walk.md:5]:
```suggestion
[Fixed Issue] Fix `AiExecutionModificationResponseList` type to correctly represent an array of responses or errors.
```

File: .changeset/yellow-mails-exist.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sap-ai-sdk/ai-api': minor
+---
+
+[Improvement] Add a function `kubesubmitV4ResourceQuotaGetDeploymentQuota()` in `DeploymentApi`to get details about quota and usage for deployments. Additionally create two new types `BckndDeploymentQuotaItem` and `BckndDeploymentResourceQuotaResponse`.
```
Comment (deekshas8) [inline .changeset/yellow-mails-exist.md:5]:
```suggestion
[Improvement] Add `kubesubmitV4ResourceQuotaGetDeploymentQuota()` function in `DeploymentApi` to get details about quota and usage for deployments. Additionally introduce two new types `BckndDeploymentQuotaItem` and `BckndDeploymentResourceQuotaResponse`.
```

**Comment by deekshas8** [comment]
@shibeshduw I would update the README in this PR as well.

**Review by deekshas8 — APPROVED** [review]

**Pull request merged**